app/src/main/java/org/thoughtcrime/securesms/util/AvatarUtil.java

```diff
@@ -0,0 +1,165 @@
+    }
+
+    GlideApp.with(target)
+            .load(photo)
+            .transform(new BlurTransformation(context, 0.25f, BlurTransformation.MAX_RADIUS))
+            .into(new CustomViewTarget<View, Drawable>(target) {
+              @Override
+              public void onLoadFailed(@Nullable Drawable errorDrawable) {
+                target.setImageDrawable(null);
+                target.setBackgroundColor(ContextCompat.getColor(target.getContext(), R.color.black));
+              }
+
+              @Override
+              public void onResourceReady(@NonNull Drawable resource, @Nullable Transition<? super Drawable> transition) {
+                target.setImageDrawable(resource);
+              }
+
+              @Override
+              protected void onResourceCleared(@Nullable Drawable placeholder) {
+                target.setImageDrawable(placeholder);
+              }
+            });
+  }
+
+  public static void loadIconIntoImageView(@NonNull Recipient recipient, @NonNull ImageView target) {
+    loadIconIntoImageView(recipient, target, -1);
+  }
+
+  public static void loadIconIntoImageView(@NonNull Recipient recipient, @NonNull ImageView target, int requestedSize) {
+    Context  context  = target.getContext();
+
+    requestCircle(GlideApp.with(context).asDrawable(), context, recipient, requestedSize).into(target);
+  }
+
+  public static Bitmap loadIconBitmapSquareNoCache(@NonNull Context context,
+                                                   @NonNull Recipient recipient,
+                                                   int width,
+                                                   int height)
+      throws ExecutionException, InterruptedException
+  {
+    return requestSquare(GlideApp.with(context).asBitmap(), context, recipient)
+                                 .skipMemoryCache(true)
+                                 .diskCacheStrategy(DiskCacheStrategy.NONE)
+                                 .submit(width, height)
+                                 .get();
+  }
+
+  @WorkerThread
+  public static IconCompat getIconForNotification(@NonNull Context context, @NonNull Recipient recipient) {
+    try {
+      return IconCompat.createWithBitmap(requestCircle(GlideApp.with(context).asBitmap(), context, recipient, UNDEFINED_SIZE).submit().get());
+    } catch (ExecutionException | InterruptedException e) {
+      return null;
+    }
+  }
+
+  @WorkerThread
+  public static @NonNull IconCompat getIconCompatForShortcut(@NonNull Context context, @NonNull Recipient recipient) {
+    try {
+      GlideRequest<Bitmap> glideRequest = GlideApp.with(context).asBitmap().load(new ConversationShortcutPhoto(recipient));
+      if (recipient.shouldBlurAvatar()) {
+        glideRequest = glideRequest.transform(new BlurTransformation(context, 0.25f, BlurTransformation.MAX_RADIUS));
+      }
+      return IconCompat.createWithAdaptiveBitmap(glideRequest.submit().get());
+    } catch (ExecutionException | InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)